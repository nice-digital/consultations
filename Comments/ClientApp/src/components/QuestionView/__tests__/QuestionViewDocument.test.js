import { createQuestionPdf , createDocumentDefinition } from "../QuestionViewDocument";
import questionsData from "./questionsData.json";
import pdfMake from "pdfmake/build/pdfmake";

jest.mock("pdfmake/build/pdfmake", ()=> {
	return {
		createPdf: jest.fn(() => {
			return {
				open: jest.fn(),
			};
		}),
	};
});

const questionsForPDF = questionsData.consultationQuestions;
const titleForPDF = questionsData.consultationTitle;
const endDate = questionsData.consultationState.endDate;

test("createPdf should be called", () => {
	createQuestionPdf(questionsForPDF, titleForPDF, endDate);
	expect(pdfMake.createPdf).toHaveBeenCalled();
});

test("document should be correctly formatted", () => {
	const definition = createDocumentDefinition(questionsForPDF, titleForPDF, endDate);
	expect(definition).toMatchSnapshot();
});
