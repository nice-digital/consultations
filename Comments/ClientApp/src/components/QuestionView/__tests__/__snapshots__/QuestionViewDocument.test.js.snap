// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document should be correctly formatted 1`] = `
Object {
  "content": Array [
    Object {
      "image": "nice-logo.png",
      "width": 250,
    },
    "
",
    "
",
    Object {
      "style": "header",
      "text": "Our test consultation title",
    },
    "
",
    Object {
      "text": "This consultation includes the following questions:",
    },
    "
",
    Object {
      "ul": Array [
        Array [
          Object {
            "text": "First Question",
          },
          "
",
        ],
      ],
    },
    "
",
    Object {
      "text": "Please note: Responses to this consultation need to be submitted online. This consultation closes on 14 December 2019.",
    },
  ],
  "defaultStyle": Object {
    "fontSize": 12,
  },
  "info": Object {
    "title": "Questions for \\"Our test consultation title\\"",
  },
  "pageMargins": 60,
  "pageOrientation": "portrait",
  "pageSize": "A4",
  "styles": Object {
    "header": Object {
      "bold": true,
      "fontSize": 16,
    },
  },
}
`;
